evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -87,17 +92,41 @@ def validate(self, X, y):
+            messages["errors"].append(DataCheckError(message="Binary class targets require exactly two unique values.",
                                                      data_check_name=self.name,
                                                      message_code=DataCheckMessageCode.TARGET_BINARY_NOT_TWO_UNIQUE_VALUES,
                                                      details=details).to_dict())
-        if self.problem_type == ProblemTypes.MULTICLASS and value_counts.min() <= 1:
-            least_populated = value_counts[value_counts <= 1]
-            details = {"least_populated_class_labels": least_populated.index.tolist()}
-            messages["errors"].append(DataCheckError(message="Target does not have at least two instances per class which is required for multiclass classification",
+
+        if self.problem_type == ProblemTypes.REGRESSION and not pd.api.types.is_numeric_dtype(y.to_series()):
+            messages["errors"].append(DataCheckError(message="Target data type should be floating point for regression type problems.",
                                                      data_check_name=self.name,
-                                                     message_code=DataCheckMessageCode.TARGET_BINARY_NOT_TWO_EXAMPLES_PER_CLASS,
-                                                     details=details).to_dict())
+                                                     message_code=DataCheckMessageCode.TARGET_UNSUPPORTED_TYPE,
+                                                     details={}).to_dict())
+
+        if self.problem_type == ProblemTypes.MULTICLASS:
+            if value_counts.min() <= 1:
+                least_populated = value_counts[value_counts <= 1]
+                details = {"least_populated_class_labels": least_populated.index.tolist()}
+                messages["errors"].append(DataCheckError(message="Target does not have at least two instances per class which is required for multiclass classification",
+                                                         data_check_name=self.name,
+                                                         message_code=DataCheckMessageCode.TARGET_BINARY_NOT_TWO_EXAMPLES_PER_CLASS,
+                                                         details=details).to_dict())
+            if len(unique_values) <= 2:
+                details = {"num_classes": len(unique_values)}
+                messages["errors"].append(DataCheckError(
+                    message="Target does not have more than two classes, which is required for multiclass classification.",
+                    data_check_name=self.name,
+                    message_code=DataCheckMessageCode.TARGET_MULTICLASS_NOT_ENOUGH_CLASSES,
+                    details=details).to_dict())
+
+            num_class_to_num_value_ratio = len(unique_values) / len(y)
+            if num_class_to_num_value_ratio >= self.multiclass_continuous_threshold:
```

Comment:
Not sure I'm understanding this. If there's 3 unique values in a list of length 50, this would warn that the problem could be regression? `3/50 == 0.06 > 0.05`. If so, this default seems really high, but definitely open to discuss.

Reply:
Good point, I think that's why this is a `warning` rather than an `error`, but maybe there could be an additional check taking into account the size of the target data when calculating this ratio?

Reply:
Yup, I think this is a good place for a discussion.  I just implemented what was in the issue as I wasn't present for the discussion - a 5% threshold.  I'm certainly down for a smarter method of doing it.